src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java

```diff
@@ -204,25 +215,28 @@ private static String getSectionBasedFieldValue(FieldType field, ShuntCompensato
         };
     }
 
-    private static String getBatteryFieldValue(FieldType field, Battery battery) {
+    private static String getBatteryFieldValue(FieldType field, String propertyName, Battery battery) {
         return switch (field) {
             case COUNTRY,
                     NOMINAL_VOLTAGE,
-                    VOLTAGE_LEVEL_ID -> getVoltageLevelFieldValue(field, battery.getTerminal().getVoltageLevel());
+                    VOLTAGE_LEVEL_ID -> getVoltageLevelFieldValue(field, null, battery.getTerminal().getVoltageLevel());
             case CONNECTED -> getTerminalFieldValue(field, battery.getTerminal());
             case MIN_P -> String.valueOf(battery.getMinP());
             case MAX_P -> String.valueOf(battery.getMaxP());
             case TARGET_P -> String.valueOf(battery.getTargetP());
             case TARGET_Q -> String.valueOf(battery.getTargetQ());
+            case FREE_PROPERTIES -> battery.getProperty(propertyName);
+            case SUBSTATION_PROPERTIES -> battery.getTerminal().getVoltageLevel().getNullableSubstation().getProperty(propertyName);
 
             default -> throw new PowsyblException(FIELD_AND_TYPE_NOT_IMPLEMENTED + " [" + field + "," + battery.getType() + "]");
 
         };
     }
 
-    private static String getSubstationFieldValue(FieldType field, Substation substation) {
+    private static String getSubstationFieldValue(FieldType field, String propertyName, Substation substation) {
         return switch (field) {
             case COUNTRY -> String.valueOf(substation.getCountry().orElse(null));
+            case FREE_PROPERTIES -> substation.getProperty(propertyName);
```

Comment:
not needed as we add a case in getFieldValue

Reply:
done